cxx-squid/src/main/java/org/sonar/cxx/CxxAstScanner.java

```diff
@@ -209,4 +221,23 @@ public SourceCode createSourceCode(SourceCode parentSourceCode, AstNode astNode)
     return builder.build();
   }
 
+  // Concatenate two strings, but if there is overlap at the intersection,
```

Comment:
... and this looks somewhat worrisome. Can you explain the reason for doing such gymnastics here ? :)